src/slave.c

```diff
@@ -193,14 +193,16 @@ void *tty_poll_loop(void *arg)
 {
     tty_state *tty = arg;
 
-    while (true) {
+    while (!tty->should_exit) {
         tty_update(tty);
         pthread_mutex_lock(&tty->lock);
         if (tty->buff_changed && tty->buff_consumed < tty->buff_len) {
             notify_ui_flush();
             tty->buff_changed = false;
         }
         pthread_mutex_unlock(&tty->lock);
+
+        // TODO: should we add small delay to avoid CPU burnout?
```

Comment:
There is no need, since in release mode event are blocking:

<img width="539" height="77" alt="Image" src="https://github.com/user-attachments/assets/9eca0569-fbf5-4bc9-ae4d-aa4f88c69a2f" />